packages/database/src/migrations/20201123061122-AddConfirmedWeeklyDataReportForPSSS-modifies-data.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+import { codeToId, insertObject, generateId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+const permissionGroupNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM permission_group WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const REPORT_CODE = 'PSSS_CONFIRMED_WEEKLY_REPORT';
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  return insertObject(db, 'report', {
+    id: generateId(),
+    code: REPORT_CODE,
+    config: {
+      fetch: {
+        dataElements: [
+          'PSSS_Confirmed_Sites_Reported',
+          'PSSS_Confirmed_Sites',
+          'PSSS_Confirmed_AFR_Cases',
+          'PSSS_Confirmed_AFR_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_DIA_Cases',
+          'PSSS_Confirmed_DIA_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_ILI_Cases',
+          'PSSS_Confirmed_ILI_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_PF_Cases',
+          'PSSS_Confirmed_PF_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_DLI_Cases',
+          'PSSS_Confirmed_DLI_Alert_Threshold_Crossed',
+        ],
+      },
+      transform: [
+        'keyValueByDataElementName',
+        {
+          transform: 'select',
+          "'period'": "convertToPeriod($row.period, 'WEEK')",
```

Comment:
Does this need to be wrapped in single quotes? Or would the following also work?

```suggestion
          "period": "convertToPeriod($row.period, 'WEEK')",
```

Reply:
So, the wrapping of keys in the `select` transform in quotes is because these keys are actually run through the `expression-parser` in order to evaluate their value. Doing this allows for some useful tricks like being able to key one value in a row by another value, eg. `"$row.dataElement" : "$row.value"`

However, this is obviously not the common case, so really we should have some logic in place to know whether or not we want to run the key through the `expression-parser`. I just haven't implemented that as of yet 😝 

I would like it do it soon tho, and if you don't wanna create any reports until we've got this done I can understand.

Reply:
No worries, it's not big deal! It's great that we have a plan for this